RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -778,7 +516,7 @@ void PFAlgo::elementLoop(const reco::PFBlock &block, reco::PFBlock::LinkData& li
 	}
       }
 
-      if ( rejectFake ) continue;
+      if ( rejectFake ) { return true; };
```

Comment:
Isn't `rejectFake` initialized as false [on L469](https://github.com/cms-sw/cmssw/blob/45a962ed4ce87e899871bb214718f03464bef678/RecoParticleFlow/PFProducer/src/PFAlgo.cc#L469), and then set possibly to `true` by `if deficit > nSigmaTRACK_*resol && !isPrimary && !goodTrackDeadHcal` on [L500](https://github.com/cms-sw/cmssw/blob/45a962ed4ce87e899871bb214718f03464bef678/RecoParticleFlow/PFProducer/src/PFAlgo.cc#L500)?

I think the if clause needs to be there on L519.

Reply:
Why? Once ```active[iTrack] = false;``` is set, and the debug line is written, there is nothing else to do inside the scope that ends on L517

I'm not saying that it is wrong to return at L519, but doing it right after L517 can avoid confusion with the following and separated definition of ```rejectFake``` (especially until the coming clang-format will re-adjust the tabbing and spaces, and then probably it will become easier to follow the flow of the code)

Reply:
Just to be sure that we are on the same page, my suggestion is

- Remove L473
```C++
  bool rejectFake = false;
```

- Modify L503-520 as
```C++
    if ( deficit > nSigmaTRACK_*resol && !isPrimary && !goodTrackDeadHcal) {
      active[iTrack] = false;
      if ( debug_ )
        std::cout << elements[iTrack] << std::endl
    	       << " deficit " << deficit << " > " << nSigmaTRACK_ << " x " << resol
    	       << " track pt " << trackRef->pt() << " +- " << trackRef->ptError()
    	       << " layers valid " << trackRef->hitPattern().trackerLayersWithMeasurement()
    	             << ", lost " << trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::TRACK_HITS)
    	             << ", lost outer " << trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::MISSING_OUTER_HITS)
    	             << ", lost inner " << trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::MISSING_INNER_HITS)
    	             << "(valid fraction " << trackRef->validFraction() <<")"
    	       << " chi2/ndf " << trackRef->normalizedChi2()
    	       << " |dxy| " << std::abs(trackRef->dxy(primaryVertex_.position())) << " +- " << trackRef->dxyError()
                   << " |dz| " << std::abs(trackRef->dz(primaryVertex_.position())) << " +- " << trackRef->dzError()
    	      << "is probably a fake (1) --> lock the track"
    	      << std::endl;
      return true;
    }
```

- Remove L523
```C++
  if ( rejectFake ) { return true; };
```

If the clause at L503 is true, all lines between 523 and 919 are skipped even now, or am I misunderstanding anything?

Reply:
Ah okay, sorry for misunderstanding, yes in this sense `rejectFake` is not really needed, I will remove it and move the return statement to the `  if ( deficit...)` block.